src/PluginClient.php

```diff
@@ -2,43 +2,24 @@
 
 namespace Http\Client\Plugin;
 
-use Http\Client\Common\EmulatedHttpAsyncClient;
-use Http\Client\Exception;
 use Http\Client\HttpAsyncClient;
 use Http\Client\HttpClient;
 use Http\Client\Plugin\Exception\LoopException;
-use Http\Promise\FulfilledPromise;
-use Http\Promise\RejectedPromise;
 use Psr\Http\Message\RequestInterface;
-use Symfony\Component\OptionsResolver\OptionsResolver;
 
 /**
  * The client managing plugins and providing a decorator around HTTP Clients.
  *
  * @author Joel Wurtz <joel.wurtz@gmail.com>
+ *
+ * @deprecated since since version 1.1, and will be removed in 2.0. Use {@link \Http\Client\Common\PluginClient} instead.
  */
 final class PluginClient implements HttpClient, HttpAsyncClient
```

Comment:
could we simply extend the base plugin client and add nothing? as the plugin interface extends the base plugin interface, everything should "just work". or is the base client final?